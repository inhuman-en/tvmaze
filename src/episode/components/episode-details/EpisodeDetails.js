import React from 'react';
import PropTypes from 'prop-types';
import SanitizedHTML from 'react-sanitized-html';
import { ErrorMessage } from 'shared';
import './EpisodeDetails.scss';

const renderImage = (data) =>
  data.image ? (
    <a className="episode-details__main__poster" href={data.image.original} rel="noreferrer" target="_blank">
      <img src={data.image.medium} alt={data.name} title={data.name} />
    </a>
  ) : (
    <div className="episode-details__main__noposter">
      <ErrorMessage text="No Image :(" />
    </div>
  );

const EpisodeDetails = ({ data }) => {
  return (
    <div className="episode-details">
      <div className="episode-details__title">{data.name}</div>
      <div className="episode-details__main">
        {renderImage(data)}
        <div className="episode-details__main__description">
          <SanitizedHTML html={data.summary || ''} allowedTags={['p', 'b']} />
        </div>
      </div>
    </div>
  );
};

EpisodeDetails.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    summary: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    image: PropTypes.shape({
      original: PropTypes.string,
      medium: PropTypes.string,
    })
  }),
};

export default EpisodeDetails;
