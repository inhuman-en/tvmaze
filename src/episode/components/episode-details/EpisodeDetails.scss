@import 'style/variables';
@import 'style/mixins';

$nav__height: 30px;
$margin: 15px;
$poster-width: 250px;
$poster-height: 140px;

.episode-details {
  display: flex;
  flex-direction: column;
  &__title {
    font-size: $font__size--large;
    font-weight: $thin;
  }

  &__main {
    display: flex;
    &__poster {
      display: block;
      margin: $margin $margin $margin 0;
      min-width: $poster-width;
      min-height: $poster-height;
      text-align: center;
    }

    &__noposter {
      display: flex;
      flex-direction: column;
      justify-content: center;
      min-width: $poster-width;
      min-height: $poster-height;
      margin: $margin $margin $margin 0;
      border: 4px dashed $color__message;
    }

    &__description {
      line-height: 20px;
      > * {
        margin: 0;
        line-height: 25px;
        text-align: justify;
      }
    }
  }

  @include md {
    align-items: center;
    &__main {
      flex-direction: column;
    }
  }
}

