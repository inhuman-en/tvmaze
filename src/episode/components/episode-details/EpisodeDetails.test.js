import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import EpisodeDetails from './EpisodeDetails';

configure({ adapter: new Adapter() });

describe('EpisodeDetails component', () => {
  const mockProps = {
    data: {
      image: {
        medium: 'https://example.com/medium',
        original: 'https://example.com/original',
      },
      name: 'episode name',
      summary: 'episode summary',
    },
  };

  it('should render', () => {
    const result = toJson(shallow(<EpisodeDetails {...mockProps} />));
    expect(result).toMatchSnapshot();
  });
});
