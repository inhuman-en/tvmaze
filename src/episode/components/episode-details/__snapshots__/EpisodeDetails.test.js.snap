// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EpisodeDetails component should render 1`] = `
<div
  className="episode-details"
>
  <div
    className="episode-details__title"
  >
    episode name
  </div>
  <div
    className="episode-details__main"
  >
    <a
      className="episode-details__main__poster"
      href="https://example.com/original"
      rel="noreferrer"
      target="_blank"
    >
      <img
        alt="episode name"
        src="https://example.com/medium"
        title="episode name"
      />
    </a>
    <div
      className="episode-details__main__description"
    >
      <SanitizedHTML
        allowedTags={
          Array [
            "p",
            "b",
          ]
        }
        html="episode summary"
      />
    </div>
  </div>
</div>
`;
